File: src/Projects/Android/ChatSystem/ChatSystemRepositoryLib/src/main/java/org/csystem/app/chatsystem/repository/IUserRepository.java
package org.csystem.app.chatsystem.repository;

import org.csystem.app.chatsystem.repository.entity.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;
@Repository
public interface IUserRepository extends CrudRepository<User, String> {
    boolean existsUserByNickNameAndPassword(@Param("nickName")String nickName, @Param("password")String password);

    Optional<User> findUserByNickNameAndPassword(@Param("nickName")String nickName, @Param("password")String password);
}

File: src/Projects/Android/PaymentApplication/RepositoryLib/src/main/java/org/csystem/android/app/payment/repository/IUserRepository.kt
package org.csystem.android.app.payment.repository

import com.karandev.util.data.repository.ICrudRepository
import org.csystem.android.app.payment.repository.entity.User

interface IUserRepository : ICrudRepository<User, String> {
    fun findByUserNameAndPassword(userName: String, password: String) : User?
    fun existsByUserNameAndPassword(userName: String, password: String) = findByUserNameAndPassword(userName, password) != null
}